vignettes/ggplot2-in-packages.Rmd

```diff
@@ -0,0 +1,247 @@
+---
+title: "Using ggplot2 in packages"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Using ggplot2 in packages}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>", fig.show = "hide")
+library(ggplot2)
+```
+
+This vignette is intended for package developers who use ggplot2 within their package code. As of this writing, this includes over 2,000 packages on CRAN and many more elsewhere! Programming with ggplot2 within a package adds several constraints, particularly if you would like to submit the package to CRAN. In particular, programming within an R package changes the way you refer to functions from ggplot2 and how you use ggplot2's non-standard evaluation within `aes()` and `vars()`.
+
+## Referring to ggplot2 functions
+
+As with any function from another package, you will have to list ggplot2 in your `DESCRIPTION` under `Imports` or `Suggests` and refer to its functions using `::` (e.g., `ggplot2::function_name`).
```

Comment:
I think it might be clearer to break this into two sections, one for imports and one for suggests.